include/nbl/asset/IAssetManager.h

```diff
@@ -219,9 +219,9 @@ class IAssetManager : public core::IReferenceCounted, public core::QuitSignallin
 
             IAssetLoader::SAssetLoadContext ctx{params, _file};
 
-            std::string filename = _file ? _file->getFileName().c_str() : _supposedFilename;
+            std::string filename = _file ? _file->getFileName().string() : _supposedFilename;
             system::IFile* file = _override->getLoadFile(_file, filename, ctx, _hierarchyLevel); // WARNING: mem-leak possibility: _override should return smart_ptr<IReadFile> (TODO, inspect this)
-            filename = file ? file->getFileName().c_str() : _supposedFilename;
+            filename = file ? file->getFileName().string() : _supposedFilename;
```

Comment:
fil the memleak issue, also use `std::filesystem::path` for the `filename` throughout consistently

Reply:
again

Reply:
@sadiuk done?